folly/CPortability.cpp

```diff
@@ -0,0 +1,581 @@
+
+#ifdef _MSC_VER
+#include <folly/CPortability.h>
+
+extern "C" {
+
+// Do nothing for the system log for now.
+void openlog(const char*, int, int) {}
+void closelog() {}
+void syslog(int, const char*, ...) {}
+
+void __stdcall __alarm_callback_func(HWND, UINT, UINT_PTR, DWORD) {
+  raise(SIGALRM);
+}
+
+unsigned int alarm(unsigned int seconds) {
+  SetTimer(NULL, 0, (UINT)(seconds * 1000), __alarm_callback_func);
```

Comment:
Is there a reason `NULL` can't be `nullptr` -- that's what we prefer. This would go for all newly added `NULL`s.